src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -342,10 +349,16 @@ internal unsafe struct MethodTable
         [FieldOffset(InterfaceMapOffset)]
         public MethodTable** InterfaceMap;
 
-        // WFLAGS_HIGH_ENUM
+        // WFLAGS_HIGH_ENUM & WFLAGS_LOW_ENUM
+        private const uint enum_flag_Category_Mask = 0x000F0000;
+        private const uint enum_flag_Category_Nullable = 0x00050000;
+        private const uint enum_flag_Category_ValueType = 0x00040000;
+        private const uint enum_flag_Category_ValueType_Mask = 0x000C0000;
         private const uint enum_flag_ContainsPointers = 0x01000000;
+        private const uint enum_flag_ComObject = 0x40000000;
         private const uint enum_flag_HasComponentSize = 0x80000000;
-        private const uint enum_flag_HasTypeEquivalence = 0x00004000;
+        private const uint enum_flag_HasDefaultCtor = 0x00000200;
+        private const uint enum_flag_HasTypeEquivalence = 0x00004000; // TODO: shouldn't this be 0x02000000?
```

Comment:
Somebody please double-check this value. I believe the value was copied from _typedesc.h_ when it really should have been copied from _methodtable.h_?

Reply:
I think none of this may be needed if you take my other suggestion to push more of the logic into one-off initialization.

Reply:
The `ActivatorCache` type checks these as part of its orchestration, since it needs to populate the backing fields differently based on whether the type is a value type, nullable, etc. Or are you suggesting that all of that logic be moved entirely into the managed world or entirely into the unmanaged world?

Reply:
There is a lot of chattiness between the managed and unmanaged sides with number of subtle internal details crossing the boundary.

I think it would be faster and look better to have a single QCall that returns everything that the activator cache needs: allocator function, allocator argument, ctor ptr and whether the ctor is public.

The part that I think makes sense to keep on the managed side is the substitution of the null allocator with a stub and the substitution of null ctor ptr with a stub. 

The COM case can be handled by allocator that takes `InteropSyncBlockInfo*`. The extra GC handle should not be needed.